Physics/Dynamics/RelativisticBreitWigner.hpp

```diff
@@ -5,149 +5,124 @@
+/// This corresponds to the Blatt Weisskopf form factors B_L like
 /// \f[
-///     B^2 = \left( \frac{q(s)}{q(s_R)} \right)^{2L+1} \times
-///                     \left( \frac{F(s)}{F(s_R)} \right)^{2}
+///     B^2 = \left( \frac{q(\sqrt{s})}{q(M_R)} \right) \times
+///           \left( \frac{M_R}{\sqrt{s}} \right) \times
+///           \left( \frac{B_L(\sqrt{s})}{B_L(\sqrt{s_R})} \right)^{2}
 /// \f]
 ///
-/// NOTE:
-/// Make sure to update the orbital angular momentum after construction,
-/// otherwise spin J will be used (which is fine is simple cases).
-class RelativisticBreitWigner
-    : public ComPWA::Physics::Dynamics::AbstractDynamicalFunction {
-
-public:
-  //============ CONSTRUCTION ==================
-  RelativisticBreitWigner(std::string name,
-                          std::pair<std::string, std::string> daughters,
-                          std::shared_ptr<ComPWA::PartList> partL);
-  virtual ~RelativisticBreitWigner();
-
-  boost::property_tree::ptree save() const;
-
-  //================ EVALUATION =================
-
-  std::complex<double> evaluate(const ComPWA::DataPoint &point,
-                                unsigned int pos) const;
-
-  /// Dynamical Breit-Wigner function.
-  /// \param mSq Invariant mass squared
-  /// \param mR Mass of the resonant state
-  /// \param ma Mass of daughter particle
-  /// \param mb Mass of daughter particle
-  /// \param width Decay width
-  /// \param L Orbital angular momentum between two daughters a and b
-  /// \param mesonRadius Meson Radius
-  /// \param ffType Form factor type
-  /// \return Amplitude value
-  static std::complex<double>
-  dynamicalFunction(double mSq, double mR, double ma, double mb, double width,
-                    unsigned int L, double mesonRadius, FormFactorType ffType);
-
-  //============ SET/GET =================
-
-  void
-  SetWidthParameter(std::shared_ptr<ComPWA::FunctionTree::FitParameter> w) {
-    Width = w;
+/// \param mSq Invariant mass squared
+/// \param mR Mass of the resonant state
+/// \param ma Mass of daughter particle
+/// \param mb Mass of daughter particle
+/// \param width Decay width
+/// \param L Orbital angular momentum between two daughters a and b
+/// \param mesonRadius Meson Radius
+/// \param ffType Form factor type
+/// \return Amplitude value
+inline std::complex<double>
+dynamicalFunction(double mSq, double mR, double ma, double mb, double width,
+                  unsigned int L, double mesonRadius, FormFactorType ffType) {
+
+  std::complex<double> i(0, 1);
+  double sqrtS = std::sqrt(mSq);
+
+  // Phase space factors at sqrt(s) and at the resonance position
+  auto phspFactorSqrtS = phspFactor(sqrtS, ma, mb);
+  auto phspFactormR = phspFactor(mR, ma, mb);
+
+  // Check if we have an event which is exactly at the phase space boundary
+  if (phspFactorSqrtS == std::complex<double>(0, 0))
+    return std::complex<double>(0, 0);
+
+  std::complex<double> qRatio =
+      (phspFactorSqrtS / phspFactormR) * mR / sqrtS;
```

Comment:
i think the equation qRatio should be : qRatio = pshpFactorSqrtS/phspFactormR, no * mR/sqrts, since pshpFactor \propto 2q/sqrtS

Reply:
Ah thats true. So just to be sure I remove the whole mR/sqrtS factor, right?

Reply:
I changed it now. Can you double check and resolve it if its correct?

Reply:
I think the original code was correct since `phspFactormR` is the phsp factor evaluated at the mass of the resonance - so mR and sqrtS do not cancel.

Reply:
The way I see it: We want to calculate `q_sqrtS/q_mR * mR/sqrtS`. That is the factor that goes in front of the Barrier Factors B_L. Since `q = phspFactor * sqrtS` (analogous for mR), `q_sqrtS/q_mR * mR/sqrtS  = phspFactorSqrtS/phspFactormR`

Reply:
No we want to calculate `q_sqrtS/q_mR` and we use the `rho_sqrtS` and `rho_mR` to do so. (for `rho` the analytic continuation is properly implemented).

Reply:
I disagree. I got my information from here: http://pdg.lbl.gov/2019/reviews/rpp2018-rev-resonances.pdf, equation 48.23. The power factor of `2L+1` will be 1 in our case, due to the different definitions of the form factors `F_L`.

Reply:
Ok I see what you mean. But the term `qRatio^(2L+1)` is meant to be `(q_sqrtS/q_mR)^(2L+1)` so the code above is correct. The additional factor `mR/SqrtS` is not exponentiated. And was added to the describtion in PDG2018 or so. Given the PDG2014 reference it is correct but if course we should update it.

Reply:
Afaik this `2L+1` is the mistake that was pointed out by @zhang-jingqing . Since his corrections are not merged yet. I would have just done that. The main problem is, that we use the formula from the pdg, but not the form factors of the pdg. Thats where these "other" equations in our code stem from

Reply:
What is the difference between the form factors we use and those suggested by the PDG?

Reply:
The form factors we used are same to the B_L(q) in DalitzAnalysis, PDG http://pdg.lbl.gov/2012/reviews/rpp2012-rev-dalitz-analysis-formalism.pdf
There is also another type B'_L(q) in same reference, and they have the relation: 
B_L(q)/B_L(q_0) = (q/q_0)^L\times B'_L(q, q_0).
The F_L(q) used in Resonance in PDG is the third form., I think after some transformation 
F_L(q) can be related to B_L(q) by: q^L\times F_L(q) = B_L(q).
In SU Chung's references, he sometimes also use F_L(q) to denote the form factor, but is different to the definition used in PDG Resonance chapter.

Reply:
So is the current implementation ok now? If so can I resolve this part?

Reply:
We should keep the current implementation at least until this large pull request is merged. The would make debugging easiert. And afterwards handle changes to the form factor description in a separate pull request.